### PR TITLE
[AND-7122] Logout realm user

### PR DESCRIPTION
**Pull request by olegoreshenko**
Task: https://rockspoon.atlassian.net/browse/AND-7122
- logout user

### CODE REVIEW

**Review by tinenbruno — APPROVED** [review]

File: app/poc/src/main/java/com/example/poc/domain/ObserveUserLoggedInUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.example.poc.domain
+
+import com.example.poc.core.data.credentials.Credentials
+import com.example.poc.core.data.credentials.CredentialsRepository
+import kotlinx.coroutines.flow.Flow
+
+class ObserveUserLoggedInUseCase(
```
Comment (allan-rockspoon) [inline app/poc/src/main/java/com/example/poc/domain/ObserveUserLoggedInUseCase.kt]: If it's just for observer a model you can inject the repository direct, but it's okay.

File: datasource/streaming_realm/src/main/java/com/example/poc/datasource/streaming_realm/RealmDatabase.kt [hunk 2]
```diff
@@ -52,6 +54,15 @@ object RealmDatabase {
         )
     }
 
+    suspend fun logout() {
+        withTimeout(2000) {
```
Comment (allan-rockspoon) [inline datasource/streaming_realm/src/main/java/com/example/poc/datasource/streaming_realm/RealmDatabase.kt:67]: There is no other way to do this? Maybe using realmApp.authenticationChangeAsFlow()?

**Review by allan-rockspoon — CHANGES_REQUESTED** [review]

File: datasource/streaming_realm/src/main/java/com/example/poc/datasource/streaming_realm/RealmDatabase.kt (hunk 2 above)
Comment (olegoreshenko) [reply datasource/streaming_realm/src/main/java/com/example/poc/datasource/streaming_realm/RealmDatabase.kt:67]: I tried using authenticationChangeAsFlow, but it didn't work for me. When user logged out, no events were in this flow.

File: core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRepository.kt [hunk 3]
```diff
@@ -111,4 +112,15 @@ class CredentialsRepository(
     suspend fun deleteCredentials() = withContext(externalScope.coroutineContext) {
         credentialsLocalDataSource.deleteCredentials()
     }
+
+    suspend fun logout() {
```
Comment (KirillChalenko) [inline core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRepository.kt]: why do you need a separate method for logout? Wasn't deleteCredentials() created for that purpose?

File: core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRealmDataSourceImpl.kt [hunk 4]
```diff
@@ -16,4 +16,10 @@ internal class CredentialsRealmDataSourceImpl(
             }
         }
     }
+
+    override suspend fun logout() {
```
Comment (KirillChalenko) [inline core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRealmDataSourceImpl.kt:20]:
```suggestion
    override suspend fun deleteCredentials() {
```

File: feature/settings/src/main/java/com/example/poc/settings/domain/LogoutUserUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.example.poc.settings.domain
+
+import com.example.poc.core.data.credentials.CredentialsRepository
+
+class LogoutUserUseCase(private val credentialsRepository: CredentialsRepository) {
```
Comment (KirillChalenko) [inline feature/settings/src/main/java/com/example/poc/settings/domain/LogoutUserUseCase.kt]: the class looks useless

File: core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRealmDataSourceImpl.kt (hunk 4 above)
Comment (olegoreshenko) [reply core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRealmDataSourceImpl.kt:20]: CredentialsRealmDataSourceImpl is confusing for me. It doesn't save or store anything. It just initialize Realm with credentials. Also, when we call delete, it doesn't delete, it logs out user. Maybe we should remove this DataSource and call it from usecase ?

Comment (allan-rockspoon) [reply core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRealmDataSourceImpl.kt:20]: It doesn't store directly, but Realm keeps the credentials inside it for every sync call.

File: core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRepository.kt (hunk 3 above)
Comment (allan-rockspoon) [reply core/data/src/main/java/com/example/poc/core/data/credentials/CredentialsRepository.kt]: deleteCredentials should also delete the credentials on the Realm data source (i.e.: logout on Realm).

File: feature/settings/src/main/java/com/example/poc/settings/domain/LogoutUserUseCase.kt (hunk 5 above)
Comment (allan-rockspoon) [inline feature/settings/src/main/java/com/example/poc/settings/domain/LogoutUserUseCase.kt]: Not really necessary. Use case here is optional, it's preferable just inject the repository directly.

**Review by allan-rockspoon — APPROVED** [review]

**Pull request merged**